editor/utils/mobile.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Internal dependencies
+ */
+import { BREAK_MEDIUM } from '../constants';
+
+/**
+ * Checks if we are in a mobile resolution using window.innerWidth if available
+ *
+ * @return {Boolean}  Returns true if on mobile resolution and false if on non mobile or impossible to check.
+ */
+const isMobileChecker = () => 'object' === typeof window && window.innerWidth < BREAK_MEDIUM;
+
+/**
+ * Disables isSidebarOpened on rehydrate payload if the user is on a mobile screen size.
+ *
+ * @param  {Object}  payload   rehydrate payload
+ * @param  {Boolean} isMobile  flag indicating if executing on mobile screen sizes or not
+ *
+ * @return {Object}            rehydrate payload with isSidebarOpened disabled if on mobile
+ */
+export const disableIsSidebarOpenedOnMobile = ( payload, isMobile = isMobileChecker() ) => (
+	isMobile ? { ...payload, isSidebarOpened: false } : payload
+);
+
+/**
+ * Middleware
+ */
+
+export const mobileMiddleware = () => next => action => {
```

Comment:
@aduth, @youknowriad did we agree on a standard place to put middlewares or is this PR still in progress?

Reply:
That's a good question, I did not found a place for middlewares so I placed with the mobile logic. But if we define a place or there is a place already defined. I will correct it to be in the right place.

Reply:
Given the precedents of `with-change-detection` and `with-history`—though not middlewares but reducer enhancers—it seems to me that, for the scope of this PR, `editor/utils` is the right place. My suggestion, however, for the sake of consistency, is to turn `mobile.js` into a proper `mobile` directory, including its test file and a README in the same mold as [wCD](https://github.com/WordPress/gutenberg/blob/499ab1b2ca905e1ac35708cf57dcc87556b6c4ee/editor/utils/with-change-detection/README.md) and [wH](https://github.com/WordPress/gutenberg/blob/499ab1b2ca905e1ac35708cf57dcc87556b6c4ee/editor/utils/with-history/README.md).